File: Introductiion To Recursion/Rope Cutting Problem/solutiongreedy.py
# Function to implement the greedy rope cutting algorithm
def greedy_rope_cut(rope_length, desired_lengths):
    # Sort the desired lengths in descending order
    desired_lengths.sort(reverse=True)

    cuts = 0
    remaining_length = rope_length

    # Iterate through desired lengths
    for length in desired_lengths:
        # Check if remaining length is greater than or equal to desired length
        while remaining_length >= length:
            # Cut the rope at the desired length
            remaining_length -= length
            # Increment cuts counter
            cuts += 1

    # Return the total number of cuts
    return cuts

# Example usage
rope_length = 10
desired_lengths = [3, 5, 7]
min_cuts = greedy_rope_cut(rope_length, desired_lengths)
print("Minimum cuts using greedy algorithm:", min_cuts)

File: Introductiion To Recursion/Rope Cutting Problem/solutiongreedy.js
// Function to implement the greedy rope cutting algorithm
function greedyRopeCut(ropeLength, desiredLengths) {
    // Sort the desired lengths in descending order
    desiredLengths.sort((a, b) => b - a);

    let cuts = 0;
    let remainingLength = ropeLength;

    // Iterate through desired lengths
    for (let length of desiredLengths) {
        // Check if remaining length is greater than or equal to desired length
        while (remainingLength >= length) {
            // Cut the rope at the desired length
            remainingLength -= length;
            // Increment cuts counter
            cuts++;
        }
    }

    // Return the total number of cuts
    return cuts;
}

// Example usage
let ropeLength = 10;
let desiredLengths = [3, 5, 7];
let minCuts = greedyRopeCut(ropeLength, desiredLengths);
console.log("Minimum cuts using greedy algorithm:", minCuts);
